platform/viewer/src/appInit.js

```diff
@@ -17,6 +17,7 @@ import {
   errorHandler
   // utils,
 } from '@ohif/core';
+import ConfigPoint from 'config-point';
```

Comment:
Taking a quick look, I saw you are using config-point, is this intended or mistakenly put here? since we haven't reviewed the config-point additions before

Reply:
It is intended to be used - it allows a site to reasonably define application specific extensions.

Reply:
I don't think this the best approach. This PR currently adds the `config-points` dependencies (which is an external 3rd party package dependency on your Github) to a lot of core code. First of all, I think if we are going ahead with `config-point` which we were going to discuss, and it should have its own PR; for transparency and maintainability, it should be an OHIF package (something like platform/config-point). The same is true for the static-wado stuff which we need to move to @cornerstone (which I think we have talked about it previously).

Hanging protocols are somehow flexible right now, and they can be set for specific user (for instance fetching from a server user-specific HP's), and they can be added in the ModeEnter to the HangingProtocolService (`HangingProtocolService.setHangingProtocol()`), so extensions are not the only place providing the protocols for matching, Mode's can manipulate the HP (and that is the intension of Modes, PT/CT mode: 3x3 layout, Prostate MRI 2x2 layout etc.). What is config-point doing for enhancement of HP stuff?

CC @swederik 

Reply:
I'm fine with config-point having it's own PR, but it can't be an OHIF package since it isn't OHIF specific, but is intended to be used in other contexts (in fact I'm already using it elsewhere).  
modes are defined in extensions, so whether the additional HPs are in the mode or are in the base extension, they are still defined elsewhere, and that means modifying code in order to provide them.  The design for ConfigPoint is that it is an easily included configuration system that allows having several variations on the configuration, without needing to include services that are not commonly available, such as the DICOM Hanging Protocol service - which exists in a couple of places but is very cumbersome to use and will mostly require sites adding custom code in order to support it.  config-point just needs static files added, so it meets the same sort of requirements for deployment that you are going to have for OHIF anyways.  The files can be gzipped on disk and served already compressed, they can be edited in any plain text editor and redeployed without rebuilding any of the software.  That is being used already for specific configurations deployed to different teams within a healthcare setting (FU Berlin specifically, different deployments for each department, and for each species being treated).
In terms of use with modes, I actually agree there - the modes should have names/settings for specific use cases, but should also be site definable - not require code level customizations to setup.  If a site wants the PT/CT mode to auto-open a fusion viewport, and then two other viewports, well, that becomes configurable at the site, as long as the underlying code has it supported.

Reply:
I'm fine with config-point having it's own PR, but it can't be an OHIF package since it isn't OHIF specific, but is intended to be used in other contexts (in fact I'm already using it elsewhere).
modes are defined in extensions, so whether the additional HPs are in the mode or are in the base extension, they are still defined elsewhere, and that means modifying code in order to provide them. The design for ConfigPoint is that it is an easily included configuration system that allows having several variations on the configuration, without needing to include services that are not commonly available, such as the DICOM Hanging Protocol service - which exists in a couple of places but is very cumbersome to use and will mostly require sites adding custom code in order to support it. config-point just needs static files added, so it meets the same sort of requirements for deployment that you are going to have for OHIF anyways. The files can be gzipped on disk and served already compressed, they can be edited in any plain text editor and redeployed without rebuilding any of the software. That is being used already for specific configurations deployed to different teams within a healthcare setting (different deployments for each department, and for each species being treated).
In terms of use with modes, I actually agree there - the modes should have names/settings for specific use cases, but should also be site definable - not require code level customizations to setup. If a site wants the PT/CT mode to auto-open a fusion viewport, and then two other viewports, well, that becomes configurable at the site, as long as the underlying code has it supported.

Reply:
we can resolve this i guess after the new commits